### PR TITLE
feat(inputs.airthings): Add Airthings plugin

### LINKED ISSUE #11878: feat: New Airthings input plugin
### Use Case

Airthings (https://www.airthings.com/) is supplying an API that allows you to access the data collected from sensors that indoor air quality and mainly the gas radon. 
We would like to have a Telegraf input plugin that exposes Airthings metrics
Expose metrics for Airthings in Telegraf collector to enable processing and aggregation of Airthings sensor data.

### Expected behavior

In the near future, I will propose PR which adds a new plugin I have created. The plugin will be able to fetch Airthings metrics from the Airthings API.

### Actual behavior

Currently, it is not possible to gather and expose Airthings metrics.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by marcgeld**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format]

resolves #11878

Adds the Airthings consumer API access to Telegraf.

Airthings REST API developer docs: https://developer.airthings.com/docs/api-getting-started/index.html 
Info about the the Airthings products: https://www.airthings.com/.

### CODE REVIEW

File: plugins/inputs/all/airthings.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+//go:build !custom || inputs || inputs.aerospike
```
Comment (srebhan) [inline plugins/inputs/all/airthings.go:1]: I don't think you are `aerospike`!? ;-)

**Review by srebhan — COMMENTED** [review]
Thanks @marcgeld for the nice plugin! The code is a good starting-point, even though I do have a few comments (see the code). I like to ask you to make use of Telegraf's HTTPClient implementation as much as possible.

Furthermore, please think about _test-case_ based tests similar to the ones in the [`ntpq` plugin](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/ntpq). In your case you would specify the input JSONs together with a target path as input and the expected output in line-protocol (similar to what is done in `ntpq`). This way it is dead-simple to add new tests by just copy over the data and config of a user and specify the expected outcome/error.

**Comment by marcgeld** [comment]
Thanks @srebhan for looking into the code almost right away, impressive!! This is my first Telegraf plug-in and I am thankful for every improvement! I will go through your comments and change my code accordingly.

**Comment by srebhan** [comment]
@marcgeld any update on the code?

**Comment by srebhan** [comment]
@marcgeld looking forward to review your updates...

**Comment by marcgeld** [comment]
> @marcgeld looking forward to review your updates...

Yes, it has been slow. Thank you for your patience. I have received new Airthings devices that I wanted to test that they work with what I have already done. I will try to finish this as soon as I can, so that the changes can be reviewed.

**Comment by powersj** [comment]
@marcgeld - thanks for putting this up. The next minor release of Telegraf, v1.26.0, is coming up in March. I would love to get your new plugin released then!

Do you think you could do the changes that Sven requested soon? And then we could add additional features/PRs down the road?

Thanks!

**Comment by marcgeld** [comment]
I'll try. time is a limiting factor, but I will try to prioritize the changes to meet the March release.

**Comment by powersj** [comment]
@marcgeld - any updates on resolving the review comments?

Thanks!

**Comment by marcgeld** [comment]
> @marcgeld - any updates on resolving the review comments?
> 
> Thanks!

Yep, I'm on it. But not yet finished. Unfortunately is time a limiting factor for me...

File: plugins/inputs/airthings/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- segment.active
+- segment.id
+- segment.name
+- segment.started
+
+## Example Output
+
+Here is example output of this plugin:
+
+```shell
```
Comment (Hipska) [inline plugins/inputs/airthings/README.md:97]: Not a shell script
```suggestion
```text
```

**Review by Hipska — COMMENTED** [review]
Would you mind to mark this PR as draft in the meantime, so we know to review again when done?

**Comment by powersj** [comment]
Hi,

v1.27 will release Monday, June 12, less than 2 weeks away. Any updates? 

As this has sat for quite a while, I am tempted to suggest that we close the PR and you can re-open once you have had time to circle back around. Would still love to see this land.

Thanks

**Pull request closed without merging**